Mathlib/Data/Set/Prod.lean

```diff
@@ -93,17 +93,15 @@ theorem prod_univ {s : Set α} : s ×ˢ (univ : Set β) = Prod.fst ⁻¹' s := b
 @[simp] lemma prod_eq_univ [Nonempty α] [Nonempty β] : s ×ˢ t = univ ↔ s = univ ∧ t = univ := by
   simp [eq_univ_iff_forall, forall_and]
 
-@[simp]
```

Comment:
This change is gonna break a lot of things (and it looks like CI agrees). But I agree it's a good change. Let's make it in a separate PR though.

Reply:
#22937